src/Components/Publishing/Ads/AdScript.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from "react"
+import { data as sd } from "sharify"
+import { ArticleAdType, ArticleData } from "../Typings"
+
+interface Props {
+  article: ArticleData
+}
+
+export class AdScript extends Component<Props> {
+  getArticleType(article: ArticleData) {
+    switch (true) {
+      case article.layout === "news":
+        return ArticleAdType.NewsLanding
+
+      case article.layout === "feature" && !this.isSponsored(article):
+        return ArticleAdType.Feature
+
+      case article.layout === "feature" && this.isSponsored(article):
+        return ArticleAdType.SponsorFeature
+
+      case (article.layout === "series" || article.layout === "video") &&
+        this.isSponsored(article):
+        return ArticleAdType.SponsorLanding
+
+      default:
+        return ArticleAdType.Article
+    }
+  }
+  isSponsored(article: ArticleData) {
```

Comment:
#minor but can we add line breaks in between each of these class methods so things don't all run together?